src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4041,6 +4076,32 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
             {
                 analyzedArguments.Free();
             }
+
+            static void validateRecordCopyConstructor(NamedTypeSymbol containingType, MethodSymbol constructor, NamedTypeSymbol baseType, MethodSymbol baseCtor, Location errorLocation, DiagnosticBag diagnostics)
+            {
+                // Are we dealing with a user-defined copy constructor in a record type?
+                if (containingType is SourceNamedTypeSymbol sourceType &&
+                    sourceType.IsRecord() &&
+                    SynthesizedRecordCopyCtor.FindCopyConstructor(containingType).Equals(constructor, TypeCompareKind.ConsiderEverything))
+                {
+                    if (baseType.SpecialType == SpecialType.System_Object)
```

Comment:
>if (baseType.SpecialType == SpecialType.System_Object) [](start = 20, length = 54)

Why even call this function under this condition?

Reply:
Simplifies the control flow of the calling method.

---
In reply to: [438505643](https://github.com/dotnet/roslyn/pull/45022#discussion_r438505643) [](ancestors = 438505643)